examples/data_type.rs

```diff
@@ -44,6 +45,13 @@ unsafe extern "C" fn free(value: *mut c_void) {
     drop(Box::from_raw(value.cast::<MyType>()));
 }
 
+unsafe extern "C" fn digest(md: *mut raw::RedisModuleDigest, value: *mut c_void) {
+    let mut dig = Digest::new(md);
+    let val = &*(value.cast::<MyType>());
+    dig.add_string_buffer(&val.data.as_bytes());
+    dig.end_sequence();
```

Comment:
so we exercise `add_string_buffer` and `end_sequence` via Integration Tests.  How can we add tests to also test `get_key_name`, `get_db_id` and `add_long_long`?

Reply:
Thanks for calling this out. I'm thinking that we can follow [core]( https://github.com/valkey-io/valkey/blob/b0f23df16522e91a769c75646166045ae70e8d4e/tests/modules/datatype2.c#L686) and integration test `get_key_name`, `get_db_id` and `add_long_long`

Reply:
Thanks for calling this out. I'm thinking that we can follow [core](https://github.com/valkey-io/valkey/blob/b0f23df16522e91a769c75646166045ae70e8d4e/tests/modules/datatype2.c#L620) and integration test `get_key_name`, `get_db_id` and `add_long_long`

Reply:
@KarthikSubbarao - your thoughts?

Reply:
Yes, it makes sense to test all the new functionality that being added, so I agree with your comment, @dmitrypol .


@hahnandrew That makes sense. We can follow the core and have the Digest callback of the example data type use all the new wrapper Digest APIs added. 

Here is how the core tests the Module APIs (using the core example module).

Core Unit Tests written in TCL: https://github.com/valkey-io/valkey/blob/2df56d87c0ebe802f38e8922bb2ea1e4ca9cfa76/tests/unit/moduleapi/datatype2.tcl#L183

Core Module: https://github.com/valkey-io/valkey/blob/b0f23df16522e91a769c75646166045ae70e8d4e/tests/modules/datatype2.c#L614

For the `get_db_id`, notice how they use `SELECT` to switch data bases. We can do the same and validate that the APIs work as expected